services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/docker_compose_utils.py

```diff
@@ -1,38 +1,52 @@
 """ Wrapper around docker-compose CLI with pre-defined options
 
 
+docker_compose_* coroutines are implemented such that they can only
+run sequentially by this service
+
 """
 import logging
+from copy import deepcopy
+from pprint import pformat
 from typing import Optional
 
-from .settings import DynamicSidecarSettings
+from servicelib.async_utils import run_sequentially_in_context
+
+from .settings import ApplicationSettings
 from .utils import CommandResult, async_command, write_to_tmp_file
 
 logger = logging.getLogger(__name__)
 
 
-async def _write_file_and_run_command(
-    settings: DynamicSidecarSettings,
-    compose_spec_yaml_content: str,
+@run_sequentially_in_context()
```

Comment:
This might not work as expected in all situations. It takes into consideration the input arguments. So two different docker-compose commands will still be able to run in parallel. (like `rm` and `up`).

For sure this will guarantee only 1 up or rm commands will be ran at once. But an up and rm can still go in parallel.

Reply:
reviewed together ... it will enforce sequential execution regardless of arguments